libvast/src/table_slice.cpp

```diff
@@ -359,15 +359,22 @@ std::shared_ptr<arrow::RecordBatch> as_record_batch(const table_slice& slice) {
       //                 == table_slice::encoding::arrow) { ... }
       if constexpr (std::decay_t<decltype(*state(encoded, slice.state_))>::encoding
                     == table_slice::encoding::arrow) {
-        return state(encoded, slice.state_)->record_batch();
+        // Get the record batch first, then create a copy that shares the
+        // lifetime with the chunk.
+        auto batch = state(encoded, slice.state_)->record_batch();
+        auto result = std::shared_ptr<arrow::RecordBatch>{
+          batch.get(), [slice, batch](arrow::RecordBatch*) noexcept {
+            VAST_IGNORE_UNUSED(slice);
+            VAST_IGNORE_UNUSED(batch);
```

Comment:
Capturing `batch` prolongs the lifetime of the `arrow::Buffer` object, correct?

Reply:
Yes, that is correct.

Reply:
I think that should be documented in a comment.

Reply:
Will do.